test/e2e/attractor_test.go

```diff
@@ -0,0 +1,110 @@
+					IPFamily: "DualStack",
+				},
+			}
+			Expect(kubeAPIClient.Create(context.Background(), trench)).Should(Succeed())
+			replicas := new(int32)
+			*replicas = 1
+			attractor = &meridiov1alpha1.Attractor{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      attractorName,
+					Namespace: namespace,
+					Labels: map[string]string{
+						"trench": trenchName,
+					},
+				},
+				Spec: meridiov1alpha1.AttractorSpec{
+					VlanID:        100,
+					VlanInterface: "eth0",
+					Replicas:      replicas,
+					Gateways:      []string{},
+					Vips:          []string{},
+				},
+			}
+			Expect(kubeAPIClient.Create(context.Background(), attractor)).Should(Succeed())
+		})
+
+		AfterEach(func() {
+			Expect(kubeAPIClient.Delete(context.Background(), attractor)).Should(Succeed())
+			Expect(kubeAPIClient.Delete(context.Background(), trench)).Should(Succeed())
+		})
+
+		It("should have the attractor pods in running state", func() {
+			By("checking if lb-fe pods are in running state")
+			loadBalancerName := fmt.Sprintf("lb-fe-%s", trench.ObjectMeta.Name)
+			Eventually(func() bool {
+				deployment, err := clientset.AppsV1().Deployments(namespace).Get(context.Background(), loadBalancerName, metav1.GetOptions{})
+				if err != nil {
+					return false
+				}
+				return deployment.Status.ReadyReplicas == deployment.Status.Replicas
+			}, timeout, interval).Should(BeTrue())
+			listOptions := metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("app=%s", loadBalancerName),
+			}
+			pods, err := clientset.CoreV1().Pods(namespace).List(context.Background(), listOptions)
+			Expect(err).ToNot(HaveOccurred())
+			deployment, err := clientset.AppsV1().Deployments(namespace).Get(context.Background(), loadBalancerName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(pods.Items)).To(Equal(int(deployment.Status.Replicas)))
+			for _, pod := range pods.Items {
+				Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+			}
+
+			By("checking if nse-vlan pods are in running state")
+			nseVLANName := fmt.Sprintf("nse-vlan-%s", attractorName)
+			Eventually(func() bool {
+				deployment, err := clientset.AppsV1().Deployments(namespace).Get(context.Background(), nseVLANName, metav1.GetOptions{})
+				if err != nil {
+					return false
+				}
+				return deployment.Status.ReadyReplicas == deployment.Status.Replicas
+			}, timeout, interval).Should(BeTrue())
+			listOptions = metav1.ListOptions{
+				LabelSelector: fmt.Sprintf("app=%s", nseVLANName),
+			}
+			pods, err = clientset.CoreV1().Pods(namespace).List(context.Background(), listOptions)
+			Expect(err).ToNot(HaveOccurred())
+			deployment, err = clientset.AppsV1().Deployments(namespace).Get(context.Background(), nseVLANName, metav1.GetOptions{})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(len(pods.Items)).To(Equal(int(deployment.Status.Replicas)))
+			for _, pod := range pods.Items {
+				Expect(pod.Status.Phase).To(Equal(corev1.PodRunning))
+			}
```

Comment:
IMO, this is also testing kubernetes. In the following quoted line(L89), I suppose it is checking if all the pods are in ready status, now it checks if the pods of the deployment are actually running. So validates the consistency in kubernetes deployment status and pod status, but not testing the operator itself

> 				return deployment.Status.ReadyReplicas == deployment.Status.Replicas

Reply:
Right, there could be crashes after applying a CR. But checking if the pod is still running cannot always catch this because the pods can be restarted. Do you know about containerStatuses.restartCount? Is this something can test this case?